crates/orchestrator/src/tests/jobs/da_job/mod.rs

```diff
@@ -84,32 +111,113 @@ async fn test_process_job() {
             nonces: vec![],
         },
     });
+
+    let pending_state_update = serde_json::to_value(&pending_state_update).unwrap();
+    let response = json!({ "id": 1,"jsonrpc":"2.0","result": pending_state_update });
+
+    let state_update_mock = server.mock(|when, then| {
+        when.path("/").body_contains("starknet_getStateUpdate");
+        then.status(200).body(serde_json::to_vec(&response).unwrap());
+    });
+
+    let response = DaJob
+        .process_job(
+            config.as_ref(),
+            &mut JobItem {
+                id: Uuid::default(),
+                internal_id: internal_id.to_string(),
+                job_type: JobType::DataSubmission,
+                status: JobStatus::Created,
+                external_id: ExternalId::String("1".to_string().into_boxed_str()),
+                metadata: HashMap::default(),
+                version: 0,
+            },
+        )
+        .await;
+
+    assert_matches!(response,
+        Err(e) => {
+            let expected_error = eyre!(
+                "Cannot process block {} for job id {} as it's still in pending state",
+                internal_id.to_string(),
+                Uuid::default()
+            )
+            .to_string();
+            assert_eq!(e.to_string(), expected_error);
+        }
+    );
+
+    state_update_mock.assert();
+}
+
+/// Tests successful DA Job processing with valid state update and nonces files.
+/// Mocks DA client to simulate environment and expects job to process without errors.
+/// Validates the successful job processing by checking the return message "Done".
+/// Asserts correct behavior by comparing the received and expected success messages.
+#[rstest]
+#[case(
+    "src/tests/jobs/da_job/test_data/state_update/631861.txt",
+    "src/tests/jobs/da_job/test_data/nonces/631861.txt",
+    "631861"
+)]
+#[case(
+    "src/tests/jobs/da_job/test_data/state_update/640641.txt",
+    "src/tests/jobs/da_job/test_data/nonces/640641.txt",
+    "640641"
+)]
+#[case(
+    "src/tests/jobs/da_job/test_data/state_update/638353.txt",
+    "src/tests/jobs/da_job/test_data/nonces/638353.txt",
+    "638353"
+)]
+#[tokio::test]
+async fn test_da_job_process_job_success(
+    #[case] state_update_file: String,
+    #[case] nonces_file: String,
+    #[case] internal_id: String,
+) {
+    // Mocking DA client calls
+    let mut da_client = MockDaClient::new();
+    da_client.expect_publish_state_diff().with(always(), always()).returning(|_, _| Ok("Done".to_string()));
+    da_client.expect_max_blob_per_txn().with().returning(|| 6);
+    da_client.expect_max_bytes_per_blob().with().returning(|| 131072);
```

Comment:
as someone not completely familar with the whole codebase I'm not sure where these numbers come from?

Reply:
Valid, now using constants defined in `orchestrator/src/tests/common.rs`


### IMPORTANT 
[This branch is a fork from [increasing coverage](https://github.com/madara-alliance/madara-orchestrator/tree/feat/increasing-test-coverage) ]
[This comment refers to changed already merged to main, they are shown here since #52 is merged and the base of this PR automatically shifted to `main` ]